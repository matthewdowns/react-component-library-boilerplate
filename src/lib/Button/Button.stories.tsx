import React, { Fragment } from 'react';
import Button from './Button';

export default {
    title: Button.name,
    component: Button,
    parameters: {
        actions: {
            handles: ['click']
        }
    }
}

export const Default = (): JSX.Element => (
    <Button>
        Button
    </Button>
);

export const Block = (): JSX.Element => (
    <Button block={true}>
        Block
    </Button>
);

export const Border = (): JSX.Element => (
    <Button>
        Border
    </Button>
)

export const Disabled = (): JSX.Element => (
    <Button disabled={true}>
        Disabled
    </Button>
);

export const Shape = (): JSX.Element => (
    <Fragment>
        <Button shape="square">
            Square
        </Button>
        <Button shape="round">
            Round
        </Button>
        <Button shape="circle">
            Circle
        </Button>
    </Fragment>
);

export const Size = (): JSX.Element => (
    <Fragment>
        <Button size="small">
            Small
        </Button>
        <Button size="medium">
            Medium
        </Button>
        <Button size="large">
            Large
        </Button>
    </Fragment>
);

export const Variant = (): JSX.Element => (
    <Fragment>
        <Button variant="primary">
            Primary
        </Button>
        <Button variant="secondary">
            Secondary
        </Button>
    </Fragment>
)
