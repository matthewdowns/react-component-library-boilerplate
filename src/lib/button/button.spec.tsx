import { render } from '@testing-library/react'
import React from 'react'
import Button from './button'
import ButtonProps from './button.props'

const children = 'Button'
const testid = 'button'

describe('Button', () => {
    it('matches snapshot', () => {
        const body = renderButton({ children })
        expect(body.container).toMatchSnapshot()
    })

    it('renders props:children', () => {
        const body = renderButton()
        const button = body.getByTestId(testid)
        expect(button).toHaveTextContent(children)
    })

    it('renders props:block', () => {
        const body = renderButton({ block: true })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-block', 'true')
    })

    it('renders props:disabled', () => {
        const body = renderButton({ disabled: true })
        const button = body.getByTestId(testid)
        expect(button).toBeDisabled()
    })

    it('renders props:hollow', () => {
        const body = renderButton({ hollow: true })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-hollow', 'true')
    })

    it('renders props:shape:square', () => {
        const body = renderButton({ shape: 'square' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-shape', 'square')
    })

    it('renders props:shape:round', () => {
        const body = renderButton({ shape: 'round' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-shape', 'round')
    })

    it('renders props:shape:circle', () => {
        const body = renderButton({ shape: 'circle' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-shape', 'circle')
    })

    it('renders props:size:small', () => {
        const body = renderButton({ size: 'small' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-size', 'small')
    })

    it('renders props:size:medium', () => {
        const body = renderButton({ size: 'medium' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-size', 'medium')
    })

    it('renders props:size:large', () => {
        const body = renderButton({ size: 'large' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-size', 'large')
    })

    it('renders props:variant:primary', () => {
        const body = renderButton({ variant: 'primary' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-variant', 'primary')
    })

    it('renders props:variant:secondary', () => {
        const body = renderButton({ variant: 'secondary' })
        const button = body.getByTestId(testid)
        expect(button).toHaveAttribute('data-variant', 'secondary')
    })
})

const defaultProps: ButtonProps = {
    children,
    'data-testid': testid
}

function renderButton(props?: Partial<ButtonProps>) {
    return render(
        <Button
            {...defaultProps}
            {...props}
        />
    )
}
