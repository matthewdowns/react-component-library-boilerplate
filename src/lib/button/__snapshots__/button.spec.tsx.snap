// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button
    class="button"
    data-shape="square"
    data-size="medium"
    data-testid="button"
    data-variant="primary"
  >
    Button
  </button>
</div>
`;
